tests/crc_test.c

```diff
@@ -99,6 +102,18 @@ static int s_test_crc32c(struct aws_allocator *allocator, void *ctx) {
     res |= s_test_known_crc32c(CRC_FUNC_NAME(aws_checksums_crc32c));
     res |= s_test_known_crc32c(CRC_FUNC_NAME(aws_checksums_crc32c_sw));
 
+
+    struct aws_byte_buf avx_buf;
+    /* enough for two avx512 runs */
+    aws_byte_buf_init(&avx_buf, allocator, 512);
+    aws_device_random_buffer(&avx_buf);
+
+    uint32_t crc = aws_checksums_crc32c_sw(avx_buf.buffer, (int)avx_buf.len, 0);
+    uint32_t hw_crc = aws_checksums_crc32c_hw(avx_buf.buffer, (int)avx_buf.len, 0);
+
+    aws_byte_buf_clean_up(&avx_buf);
+    ASSERT_UINT_EQUALS(hw_crc, crc);
```

Comment:
This should test various sizes of unaligned buffers as well. I'd prefer to test every buffer size from 0 to 512+1 bytes, and have some tests that deliberatly unalign the buffer, and/or put it just before a page boundary where the next page is inaccessible.